ci-operator/step-registry/ipi/conf/aws/ipi-conf-aws-commands.sh

```diff
@@ -31,6 +31,7 @@ if [[ "${SIZE_VARIANT}" == "compact" ]]; then
   workers=0
 fi
 master_type=null
+COMPUTE_NODE_TYPE=null
```

Comment:
```console
$ grep -3 COMPUTE_NODE_TYPE ci-operator/step-registry/ipi/conf/aws/ipi-conf-aws-commands.sh
# Generate working availability zones from the region
mapfile -t AVAILABILITY_ZONES < <(aws --region "${REGION}" ec2 describe-availability-zones | jq -r '.AvailabilityZones[] | select(.State == "available") | .ZoneName' | sort -u)
# Generate availability zones with OpenShift Installer required instance types
mapfile -t INSTANCE_ZONES < <(aws --region "${REGION}" ec2 describe-instance-type-offerings --location-type availability-zone --filters Name=instance-type,Values="${BOOTSTRAP_NODE_TYPE}","${master_type}","${COMPUTE_NODE_TYPE}" | jq -r '.InstanceTypeOfferings[].Location' | sort -u)
# Generate availability zones based on these 2 criterias
mapfile -t ZONES < <(echo "${AVAILABILITY_ZONES[@]}" "${INSTANCE_ZONES[@]}" | sed 's/ /\n/g' | sort -R | uniq -d)
# Calculate the maximum number of availability zones from the region
--
  replicas: ${workers}
  platform:
    aws:
      type: ${COMPUTE_NODE_TYPE}
EOF
/tmp/yq m -x -i "${CONFIG}" "${PATCH}"
```

`COMPUTE_NODE_TYPE` is from #14350.  I'm pretty sure just setting it to `null` isn't what we want, because it would break use cases that  depend on it being a functioning variable.  What we want is smarter handling in the `InstanceTypeOfferings` `jq`, so we get a list of zones that support all the requested instance types, not the current list of zones that support at least one of the requested instance types.

Reply:
@wking thanks for the feedback it was really helpful. https://github.com/openshift/release/pull/14350 looks like it sets `COMPUTE_NODE_TYPE` default to `null`.  Our `COMPUTE_NODE_TYPE`  is **m4.xlarge** https://github.com/openshift/release/blob/master/ci-operator/config/integr8ly/integreatly-operator/integr8ly-integreatly-operator-master.yaml#L101 and that returns all zones  

```bash
aws --region "us-east-1" ec2 describe-instance-type-offerings --location-type availability-zone --filters Name=instance-type,Values="m5.large","null","m4.xlarge" | jq -r '.InstanceTypeOfferings[].Location' | sort -u
us-east-1a
us-east-1b
us-east-1c
us-east-1d
us-east-1e
us-east-1f
```
I will play around with jq and InstanceTypeOffering to see if i can get a better solution